Add learn doc about various ways to share code (#3201)
astronomer/docs: learn/managing-airflow-code.md

## Reusing code

> Code such as custom hooks, operators, or DAG templates that are reused between projects should be stored in a repository that's separate from your individual project repositories. This ensures that any changes to the re-used code only need to be made once and are applied across all projects where that code is used. Read [Sharing code across projects](https://docs.astronomer.io/learn/sharing-code-multiple-projects) about how to structure and set up a project for shared code.

Reviewer: ```suggestion
Code such as custom hooks, operators, or DAG templates that are reused between projects should be stored in a repository that's separate from your individual project repositories. This ensures that any changes to the re-used code only need to be made once and are applied across all projects where that code is used. See [Sharing code across projects](https://docs.astronomer.io/learn/sharing-code-multiple-projects) for more on how to structure and set up a project for shared code.
```

> You can pull code from separate repositories into your Airflow projects. If you are working with Astronomer, see [Customize your image on Astronomer Software](https://docs.astronomer.io/software/customize-image#build-from-a-private-repository). Depending on your repository setup, it may also be possible to manage this with your CI/CD tool.

Reviewer: No, it's not only software. That doc just hasn't been updated in a long time :)

I think this is the most relevant doc for Astro: https://docs.astronomer.io/astro/cli/private-python-packages

> You can pull code from separate repositories into your Airflow projects. If you are working with Astronomer, see [Customize your image on Astronomer Software](https://docs.astronomer.io/software/customize-image#build-from-a-private-repository). Depending on your repository setup, it may also be possible to manage this with your CI/CD tool.

Reviewer: ```suggestion
You can pull code from separate repositories into your Airflow projects. If you are working with Astronomer, see [Install Python packages from private sources](https://docs.astronomer.io/astro/cli/private-python-packages) for Astro or [Customize your image on Astronomer Software](https://docs.astronomer.io/software/customize-image#build-from-a-private-repository) for Software. Depending on your repository setup, it may also be possible to manage this with your CI/CD tool.
```
